File: weisi/src/main/java/com/cjq/lib/weisi/protocol/BleAnalyzer.java
package com.cjq.lib.weisi.protocol;


import com.cjq.lib.weisi.util.NumericConverter;

/**
 * Created by CJQ on 2018/1/10.
 */

public class BleAnalyzer implements Analyzable {

    @Override
    public long analyzeTimestamp(byte[] src, int position) {
        return NumericConverter.int8toUInt32ByMSB(src,
                position) * 1000;
    }

    public float analyzeBatteryVoltage(byte voltage) {
        return voltage < 0
                ? voltage
                : voltage * BATTERY_VOLTAGE_COEFFICIENT;
    }

    public double analyzeRawValue(byte[] src, int position) {
        return NumericConverter.bytesToFloatByMSB(src, position);
    }
}

File: wsb/src/main/java/com/wsn/lib/wsb/protocol/BleAnalyzer.kt
package com.wsn.lib.wsb.protocol


import com.wsn.lib.wsb.util.NumericConverter
import kotlin.experimental.and

/**
 * Created by CJQ on 2018/1/10.
 */

class BleAnalyzer : Analyzable {

    override fun analyzeTimestamp(src: ByteArray, position: Int): Long {
        return NumericConverter.int8toUInt32ByMSB(src, position) * 1000
    }

    fun analyzeBatteryVoltage(voltage: Byte): Float {
        return if (voltage < 0)
            (-(voltage and 0x7F)).toFloat()
        else
            voltage * Analyzable.BATTERY_VOLTAGE_COEFFICIENT
    }

    fun analyzeRawValue(src: ByteArray, position: Int): Double {
        return NumericConverter.bytesToFloatByMSB(src, position).toDouble()
    }
}
